src/main/java/de/tum/in/www1/artemis/repository/LectureUnitRepository.java

```diff
@@ -59,7 +60,7 @@ default LectureUnit findByIdWithCompetenciesBidirectionalElseThrow(long lectureU
     }
 
     default LectureUnit findByIdWithCompetenciesElseThrow(long lectureUnitId) {
-        return findByIdWithCompetencies(lectureUnitId).orElseThrow(() -> new EntityNotFoundException("LectureUnit", lectureUnitId));
+        return findWithCompetenciesAndSlidesById(lectureUnitId).orElseThrow(() -> new EntityNotFoundException("LectureUnit", lectureUnitId));
```

Comment:
Maybe we should also adjust the method name? Otherwise, developers might not be aware that they fetch more data than expected.